### PR TITLE
0.10.0rc1

### LINKED ISSUE #1649: Update release notes and prep 0.10 release
Title says it all :rocket:

### LINKED ISSUE #1655: Bump soundfile dependency to 0.12 (when it releases)
There is a forthcoming release of python-soundfile (0.12) which is intended to clear up some packaging issues and do a better job of ensuring mp3 support in the wheels.

I see no reason not to bump our minimum version up to 0.12 when it releases so as to sidestep upstream library issues.  If the timing is right, I'd like this to make it in for the 0.10 release.  There will be no code changes from our side, excepting the modification to the package specs and testing environments.

### LINKED ISSUE #1659: Zenodo author list
This is not an issue with the code base but with our Zenodo integration.

We've just released librosa v0.10.rc0 on PyPI, and that triggered a new Zenodo object:
https://zenodo.org/record/7618817

When reviewing it i noticed that the list of authors was surprisingly small, with some important contributors missing:
* dafaronbi 
* iranroman
* juliawilkins 
* faroit
* matangover
* Embot2000
* bshall
* beasteers
* ...

What seems to happen is that Zenodo applies a threshold on the number of commits ever made to the package. But some contributions were either squashed or coauthored with someone, so they don't look like they count as much.
https://github.com/librosa/librosa/graphs/contributors

Librosa has had 100+ contributors in its history (yay!), so i understand that it's impractical to list everyone all the time. But the current rule (6 commits or more, IIUC) looks unfair to me. Without the people i just listed, we wouldn't have multichannel processing, pYIN, perceptual loudness weighting, non-A440 tuning, advanced I/O documentation, etc.

So i'm opening this issue to discuss collectively about what would be a good policy for authorship credit in Zenodo. I am not talking about `AUTHORS.md`, which is fine. Some options i can see right now, just to kickstart the debate

(1) Replace number of commits by number of added LOCs (green), or added LOCs + removed LOCs (green + red). Either set a minimum number (e.g., 100) or restrict the author list to top-K (e.g., K=10).

(2) Include everyone who has authored at least one commit since the last stable release. Exclude everyone else, even if they have contributed a lot to previous releases.

(3) Let Brian decide on who's deserving of coauthorship credit, based on various factors: code but also issues, review, comments, etc.

Let me know what you think.

### LINKED ISSUE #1668: matplotlib 3.7 breaks image comparison tests
**Describe the bug**

This is showing up in the final CI runs on #1666 , but the recent matplotlib 3.7 release is now triggering test failures on image comparison tests.

```
=================================== FAILURES ===================================
______________________________ test_specshow_vqt _______________________________
Error: Image files did not match.
  RMS Value: 38.581767218035345
  Expected:  
    /tmp/tmp3rx4i1kp/test_display.test_specshow_vqt/baseline.png
  Actual:    
    /tmp/tmp3rx4i1kp/test_display.test_specshow_vqt/result.png
  Difference:
    /tmp/tmp3rx4i1kp/test_display.test_specshow_vqt/result-failed-diff.png
  Tolerance: 
    6
___________________________ test_specshow_chromafjs ____________________________
Error: Image files did not match.
  RMS Value: 33.977947433674615
  Expected:  
    /tmp/tmp3rx4i1kp/test_display.test_specshow_chromafjs/baseline.png
  Actual:    
    /tmp/tmp3rx4i1kp/test_display.test_specshow_chromafjs/result.png
  Difference:
    /tmp/tmp3rx4i1kp/test_display.test_specshow_chromafjs/result-failed-diff.png
  Tolerance: 
    6
```
These tests previously worked on mpl 3.6.2 a few days ago.

This is only showing up with VQT and chroma/FJS, so my hunch is that something is amiss with fonts.

I'll try to get this reproducing in my local environment to sort it out.

### PR DESCRIPTION
**Pull request by bmcfee**
#### Reference Issue
- [x] Fixes #1649 
- [x] Fixes #1659 
- [x] Fixes #1655 
- [x] Fixes #1668 

#### What does this implement/fix? Explain your changes.
Updates to author list, documentation, and upstream dependencies.

### CODE REVIEW

**Comment by bmcfee** [comment]
Ok, first cut of a .zenodo.json is up: https://github.com/librosa/librosa/blob/957024cc71cb0ca735603ef3068c7f400da0bfc1/.zenodo.json

I added ORCID ids where I was reasonably confident in their accuracy.  Unfortunately, zenodo's schema doesn't allow for a github account here to facilitate downstream linkage, but this is still more attribution than we had in the first place, so I consider this a win.

Note this is still just raw ordering by add/deletes (no more time normalization).  Suggestions at this point are welcome.

File: docs/changelog.rst [hunk 1]
```diff
@@ -10,37 +10,34 @@ v0.10.0
...
 
 New features
-
     - `#1485`_ Added support for `"h"` (hours) and `"m"` (minutes) in `librosa.display.TimeFormatter`.  *Vincent Lostanlen*
     - `#1495`_ Expanded options for bandwidth calculation in `librosa.segment.recurrence_matrix` and `librosa.segment.cross_similarity`. *Tom Xi*
     - `#1529`_ Accelerated utility functions for squared magnitude `librosa.util.abs2` and `librosa.util.phasor`. *Brian McFee*
     - `#1514`_ `librosa.stft` support for pre-allocated output arrays.  *Brian McFee*
     - `#1533`_ Accelerated `librosa.util.localmax` and `librosa.util.localmin`. *Brian McFee*
-    - `#1552`_ **EXPERIMENTAL** Support for just-intonation based variable-Q transforms `librosa.vqt` and functional just system (FJS) notation. *Brian McFee*
+    - `#1552`_ **EXPERIMENTAL** Support for just-intonation based variable-Q transforms `librosa.vqt` and functional just system (FJS) notation. *Brian McFee, Vincent Lostanlen*
```
Comment (lostanlen) [inline docs/changelog.rst:18]: LOL, thanks

**Review by lostanlen — APPROVED** [review]

File: docs/changelog.rst (hunk 1 above)
Comment (bmcfee) [reply docs/changelog.rst:18]: Sorry about that one!  It's bound to happen when I visually scan the PR list for authors and don't think too hard. :\

**Comment by bmcfee** [comment]
Build fails are premature because the soundfile 0.12.1 package hasn't propagated through the pipes yet.  Will rerun in an hour or so.

**Comment by bmcfee** [comment]
Aha, conda environments are failing now because pysoundfile 0.12.1 ships as a `.conda` archive instead of `.tar.bz2`.

I'll look into resolving this on the env setup side.

**Pull request merged**